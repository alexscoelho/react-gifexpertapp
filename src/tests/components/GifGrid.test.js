import React from "react";
import "@testing-library/jest-dom";
const { GifGrid } = require("../../components/GifGrid");
const { shallow } = require("enzyme");
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

// using mocks to fake api request
describe("Pruebas en <GifGrid />", () => {
  const category = "pokemon";
  test("debe mostrarse correctamente  ", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useFetchGifs ", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://cualquier-cosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "CDB",
        url: "https://cualquier-cosa.jpg",
        title: "Cualquier cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
