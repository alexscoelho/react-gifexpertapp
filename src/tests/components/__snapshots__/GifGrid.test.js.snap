// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    pokemon
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://cualquier-cosa.jpg"
    />
    <GifGridItem
      id="CDB"
      key="CDB"
      title="Cualquier cosa"
      url="https://cualquier-cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse correctamente   1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    pokemon
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
